### ISSUE TITLE
include travis CI into the process

### ISSUE THREAD

**Issue by atherdon** [ChickenKyiv/loopback-tutor-intern-18#11]

**Comment by atherdon** [comment]
when you'll be ready to do this tasks - just start a discussion here and i'll provide more details for oyu

**Comment by JorgeRodr** [comment]
@atherdon I have enter in tras ci webpage and to give acces to this repository but this repo is not mine so I can't do that. I can send you a request if needed.

**Comment by atherdon** [comment]
try again - i give you an admin access

**Comment by JorgeRodr** [comment]
@atherdon mmm it seems all the same to me, I send you a screenshot of what I see:
![travis](https://user-images.githubusercontent.com/39463593/41500190-a9969d34-718d-11e8-9ede-ebb1bfc43f83.PNG)

**Comment by atherdon** [comment]
@JorgeRodr ok, i'll try to add travis to this repo inside my account and maybe i can add you as a team member. hold on

**Comment by atherdon** [comment]
ok, i add it: https://travis-ci.org/ChickenKyiv/loopback-tutor-intern-18
can you create a simple travis `config` file so we can test how it works

**Comment by JorgeRodr** [comment]
@atherdon  I have created a simple travis.yml with the language and the version, I've been looking through the documentation and it seems we can set env variables there too, so maybe it could be interesting to set the sentry configuration using travis

**Comment by atherdon** [comment]
so all other config variables can be stored at tarvis.yml file too? ok,
let's try. looks like interesting solution. but usually .env using at a lot
of projects. for sure, for big project this data should be ignored, so
nobody will know our keys. but for now we don't need to hide it


On 16-Jun-2018 at 07:35 PM, JorgeRodr wrote:


@atherdon <https://github.com/atherdon> I have created a simple travis.yml
with the language and the version, I've been looking through the
documentation and it seems we can set env variables there too, so maybe it
could be interesting to set the sentry configuration using travis

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/loopback-tutor-intern-18/issues/11#issuecomment-397834617>,
or mute the thread

**Comment by JorgeRodr** [comment]
Actually inside travis settings in the project we can set environment variables and only us would be allowed to see that. Maybe could be a better way, the only bad thing is that we won't send to sentry the errors that will occur during local developing.

![travissettings](https://user-images.githubusercontent.com/39463593/41507376-a7bb730a-7231-11e8-88c0-75a005b6fc15.PNG)

If you agree with this I can do it.

**Comment by atherdon** [comment]
by keeping settings at travis CI you mean inside it;s dashboard, right?
for now i'm pretty ok with this solution, because it solves a lot of basic
problem that i have in different projects.

**Comment by JorgeRodr** [comment]
Ok I did it and I commented for themoment the variables in the .env file in the code. Until our first deploy we won't know if it really works but makes sense to me.

**Comment by atherdon** [comment]
i didn't see you image in email - i got it. heroku have a similar thing for variables too.

Can you move a similar .env login into this projects?
https://github.com/ChickenKyiv/recipe-search-api
https://github.com/ChickenKyiv/recipe-api-only
https://github.com/GroceriStar/groceristar
Also check including raven inside this projects and remove an old versions and include new way

**Comment by JorgeRodr** [comment]
what do you mean with login? Also, if we want to use travis configuration we need to link them to travis as we did with the loopback intern project, don't we?

**Comment by atherdon** [comment]
for that projects i didn't plan to use settings from travis for now


On 17-Jun-2018 at 05:03 PM, JorgeRodr wrote:


what do you mean with login? Also, if we want to use travis configuration
we need to link them to travis as we did with the loopback intern project,
don't we?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/loopback-tutor-intern-18/issues/11#issuecomment-397892145>,
or mute the thread

**Comment by JorgeRodr** [comment]
mmm then I don't understand.  Recipe-search-api project is the only of those three without a .env file. I guess you want me to create it and put the sentry configuration for raven there, but I don't know what exactly I have to do with the other two projects, besides the raven middleware I mean.

**Comment by JorgeRodr** [comment]
@atherdon Hi Arthur, I'm trying to create a feature branc on this repo:

https://github.com/ChickenKyiv/recipe-search-api

I want to work on a feature branch so I can send you a pull request later but I don't have the correct permissions. It is for the raven config and middleware, I will do it the same with the other repos so you can give me permission too on those.

**Comment by atherdon** [comment]
@JorgeRodr you should receive an invite to whole organization. sorry for not checking this

**Comment by JorgeRodr** [comment]
@atherdon I accepted the invitations but still I am having problems with the access 

https://github.com/ChickenKyiv/recipe-search-api

I can not push my branch to that repo. It says:

    remote: Permission to ChickenKyiv/recipe-search-api.git denied to JorgeRodr.
    fatal: unable to access 'https://github.com/ChickenKyiv/recipe-search-api.git/': The requested URL 
    returned error: 403

Same with the other two repos.

Also, in the groceristar repo are you working on master or develop branch? Which is more advanced?

**Comment by atherdon** [comment]
i saw this issue the second time. can you fork and make a PR?

**Comment by atherdon** [comment]
check this - https://stackoverflow.com/a/7771927 mb it will help

**Comment by JorgeRodr** [comment]
I sent you a PR of recipe-api and recipe-search. For groceristar, should I create the branch from develop or master?

**Comment by atherdon** [comment]
i think gs have develop branch - let's create a development branch -
because i'm not quite remember what changes i made on it. we'll address
that later.

**Comment by atherdon** [comment]
please create your branch from master. we didn't setup a nice development
process - because i was a main developer a lot of month for it

**Comment by JorgeRodr** [comment]
@atherdon Hi Arthtur, I'm doing the thing on groceristar project, I have created a feature branch from master as you said. I've been thinking about to create an issue on groceristar project for raven middleware. It is a larger project and not only I'm deleting a lot of code where raven was called but I think I have to change the error handling with try/catch due to next() callback been called automatically and not been necessary. As I said, I think it couldbe better to treat this stuff in a seperate issue, what do you think?

**Comment by atherdon** [comment]
100% agree

**Comment by atherdon** [comment]
Just to let you know.
Right now i'm working on documentation related to search-api tests.
when i'll finish it - i'll try to improve information related to Recipe-api first release, so it'll be more easy for you to start to work on it.

**Comment by JorgeRodr** [comment]
Yeah that would be nice, thanks! :) Regarding the issue, should I create it? I mean, I don't know if a have "the power" to create issues haha.

**Comment by atherdon** [comment]
you have power to do everything with our code and repositories :)


On 24-Jun-2018 at 06:51 PM, JorgeRodr wrote:


Yeah that would be nice, thanks! :) Regarding the issue, should I create
it? I mean, I don't know if a have "the power" to create issues haha.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/loopback-tutor-intern-18/issues/11#issuecomment-399778659>,
or mute the thread
